### DISCUSSION TITLE
Fine scrubbing or similar for precise edit points

### DISCUSSION THREAD

**Discussion by hobyvh** [mifi/lossless-cut#1287] [Ideas]
While setting and moving the start/stop points for an edit are great on the timeline, it would also be nice to have some way to easily scrub forward and backward to get just the right keyframe to either move or add an edit point.

Right now if the video is very long, timeline clicks just a pixel or few away can jump to pretty different timepoints—so I don't have much control. The timeline zoom doesn't seem to help this issue much either.

**Comment by mifi** [comment]
Not sure how fine scrubbing would work. Click and drag with a modifier key to slow down seeking? Why doesn't timeline zoom help with the issue?

**Comment by hobyvh** [comment]
Some ideas for a faster fine scrubbing interface:
1. Map timeline zoom powers to number keys.
2. Map timeline zoom to scroll up/down.
3. Temporary timeline zoom using a modifier key or tool.
4. Temporary timeline zoom when click-holding or click-dragging in the timeline.
5. Popup/tooltip zoomed timeline segment with click-hold/drag in timeline. Click outside of it to dismiss.
6. For videos where only keyframes can be edit points, mode for snapping to nearest keyframe.

Ideas for course scrubbing:
1. Any of the methods highlighted in this [Swifter video](https://www.youtube.com/watch?v=H9dqW2p4pYE).
2. Displaying the timeline as a set of stills and skimming like in iMovie or Final cut.

**Reply by mifi** [reply]
Thanks. I added your suggestions to #483

> * Map timeline zoom to scroll up/down.

I beileve tihs is already implemented

**Reply by hobyvh** [reply]
Thanks.

What I'm seeing is that scrolling in any direction (up, down, left, or right with trackpad) scrolls through the video, at the current zoom level with no change in zoom level. Same precision as clicking in the timeline.

**Comment by mifi** [comment]
Oh, I think the shortcut is CTRL+wheel?

**Reply by hobyvh** [reply]
It looks like that's true on Windows and Linux but on the Mac, control+scrollwheel conflicts with the OS Accessibility setting to allow full screen zooming.

It appears that Control should be mapped to Command on MacOS but currently command+scrollwheel is the same as scrollwheel.

Side note: the visual feedback on Windows with ctrl+scroll seems jittery with edits, too minimal with no edits, and jumps in scale a bit too fast.

**Reply by mifi** [reply]
> It looks like that's true on Windows and Linux but on the Mac, control+scrollwheel conflicts with the OS Accessibility setting to allow full screen zooming.

I have now implemented a setting for this modifier key so the user can choose ctrl,shift,alt or meta

> Side note: the visual feedback on Windows with ctrl+scroll seems jittery with edits, too minimal with no edits, and jumps in scale a bit too fast.

not sure what you mean by that exactly
